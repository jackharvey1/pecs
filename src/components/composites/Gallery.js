import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import GridImage from '../atomics/GridImage';
import MenuBar from '../atomics/MenuBar';
import Button from '../atomics/Button';

const GalleryContainer = styled.div`
    position: absolute;
    width: 100vw;
    height: 100vh;

    top: 100vh

    background: white;

    transition: top 0.4s ease-in-out 0s;

    display: flex;
    flex-flow: row wrap;
    flex-grow: 1;
    align-content: flex-start;

    @media print {
        display: none;
    }
`;

const Gallery = ({ data, hideGallery, addToStoryboard, galleryTopValue }) => (
    <GalleryContainer style={{ top: galleryTopValue }}>
        <MenuBar>
            <Button onClick={hideGallery}>Cancel</Button>
        </MenuBar>
        {data.map((image, i) => (
            <GridImage
                key={i}
                src={image.src}
                alt={image.alt}
                addToStoryboard={addToStoryboard}
            />
        ))}
    </GalleryContainer>
);

Gallery.propTypes = {
    data: PropTypes.array.isRequired,
    galleryTopValue: PropTypes.string.isRequired,
    hideGallery: PropTypes.func.isRequired,
    addToStoryboard: PropTypes.func.isRequired,
};

export default Gallery;
