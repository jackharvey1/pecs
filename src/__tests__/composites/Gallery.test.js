import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Gallery from '../../components/composites/Gallery';

describe('Gallery component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(<Gallery
            data={[
                {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                },
            ]}
            galleryTopValue="0"
            hideGallery={() => null}
            addToStoryboard={() => null}
        />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('closes on cancel', () => {
        const mockHideGallery = jest.fn();
        const wrapper = mount(<Gallery
            data={[
                {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                },
            ]}
            galleryTopValue="0"
            hideGallery={mockHideGallery}
            addToStoryboard={() => null}
        />);

        const hideButton = wrapper.find('button');
        hideButton.simulate('click');

        expect(mockHideGallery).toHaveBeenCalled();
    });

    it('calls to add image to storyboard ', () => {
        const mockAddToStoryboard = jest.fn();
        const wrapper = mount(<Gallery
            data={[
                {
                    src: 'http://image.com/image.png',
                    alt: 'image',
                },
            ]}
            galleryTopValue="0"
            hideGallery={() => null}
            addToStoryboard={mockAddToStoryboard}
        />);

        const addToStoryboard = wrapper.find('img').at(0);
        addToStoryboard.simulate('click');

        expect(mockAddToStoryboard).toHaveBeenCalled();
    });
});
