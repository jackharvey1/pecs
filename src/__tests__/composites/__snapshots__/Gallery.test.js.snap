// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component renders as intended 1`] = `
.c3 {
  --width: calc(100vw/3);
  height: var(--width);
  width: var(--width);
}

.c1 {
  width: 100%;
  height: 40px;
  color: white;
  background: linear-gradient(#ff8197,#ff9bac);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: none;
  border: none;
  color: white;
  font-variant: all-small-caps;
  font-size: 15px;
  font-weight: 100;
  font-family: sans-serif;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  padding: 0 17px;
}

.c2:focus {
  outline: none;
}

.c0 {
  position: absolute;
  width: 100vw;
  height: 100vh;
  top: 100vh;
  background: white;
  -webkit-transition: top 0.4s ease-in-out 0s;
  transition: top 0.4s ease-in-out 0s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
}

@media (min-width:768px) {
  .c3 {
    --width: calc(100vw/4);
  }
}

@media (min-width:1024px) {
  .c3 {
    --width: calc(100vw/6);
  }
}

@media (min-width:1280px) {
  .c3 {
    --width: calc(100vw/8);
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
  style={
    Object {
      "top": "0",
    }
  }
>
  <div
    className="c1"
  >
    <button
      className="c2"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
  <img
    alt="image"
    className="c3"
    onClick={[Function]}
    src="http://image.com/image.png"
  />
</div>
`;
